### ISSUE TITLE
Tests for scvelo

### ISSUE THREAD

**Issue by Marius1311** [theislab/scvelo#226]
What's the state of the art of scvelo's testing system? Is it only the tests in the `test_basic.py` file so far? I think it would be really important to add comprehensive testing to scvelo given the user number we have by now. @WeilerP , is that something you are working on already?

**Comment by VolkerBergen** [comment]
agree, we need much better test coverage. I don't have the capacity right now, but will try add tests for the key funcs asap.

**Comment by WeilerP** [comment]
Good point - I was thinking the same thing. As I am going through the preprocessing part at the moment, I was planning on adding tests for that since it also helps me to better understand the code.

**Comment by Marius1311** [comment]
That's great. @michalk8 has put quite a lot of effort into writing tests for cellrank, see https://github.com/theislab/cellrank/tree/master/tests, we have ~80 % coverage by now. This is currently not shown in our badge because we have a problem with automatically updating our coverage using github actions, see https://github.com/theislab/cellrank/issues/192

In any case, maybe cellrank's testing system can be used to get some inspiration. Happy to help if anything is unclear.

**Comment by WeilerP** [comment]
Thanks, I'll take a look at it! I was thinking about using [hypothesis](https://hypothesis.readthedocs.io/en/latest/) to have robust unit tests and easily find edge cases.

**Comment by VolkerBergen** [comment]
That'd, of course, be amazing @WeilerP, if you could add some tests when already going through the code. I think just simple pytest as already implemented for the very basic pipeline should do it. But happy to disc about alternatives if advantageous.

**Comment by Marius1311** [comment]
@michalk8 , would hypothesis based testing be interesting for cellrank as well? 
https://hypothesis.readthedocs.io/en/latest/
